### PR TITLE
feat(firestore): Query explain and explain analyze

### PR DESCRIPTION
**Pull request by bhshkh**
## Description
Adding Firestore query explain and explain analyze samples. 
The build will pass once 1.17.0 has been released. I will merge this PR once https://github.com/googleapis/google-cloud-go/pull/10597 has been merged.
This has been tested locally by pointing cloud.google.com/go/firestore to local directory

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by telpirion** [comment]
Please be sure to address the errors from vet.

File: firestore/aggregate_query_explain.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	// Get plan summary
+	planSummary := explainMetrics.PlanSummary
+	fmt.Fprintln(w, "----- Indexes Used -----")
+	for k, v := range planSummary.IndexesUsed {
+		fmt.Fprintf(w, "%+v: %+v\n", k, v)
+	}
+
+	return nil
+}
+
```
Comment (telpirion) [inline firestore/aggregate_query_explain.go:58]: nit: remove blank line between final curly brace and ending region tag.

File: firestore/aggregate_query_explain_analyze.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	ctx := context.Background()
+
+	// Create client
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	// Build the query. Set explain options with analzye = true to get back the
```
Comment (telpirion) [inline firestore/aggregate_query_explain_analyze.go:37]: Same as previous sample: cut "Build the query" from this comment.

**Review by telpirion — APPROVED** [review]
Just a couple of nits, accept or reject as you see fit. Thank you!

File: firestore/aggregate_query_explain.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	ctx := context.Background()
+
+	// Create client
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	// Build the query
```
[the remark this answers is not in the document]
Comment (bhshkh) [reply firestore/aggregate_query_explain.go]: Updated

File: firestore/aggregate_query_explain_analyze.go (hunk 2 above)
Comment (bhshkh) [reply firestore/aggregate_query_explain_analyze.go]: Updated

File: firestore/aggregate_query_explain.go (hunk 1 above)
Comment (bhshkh) [reply firestore/aggregate_query_explain.go:57]: Tried removing it. gofmt adds it back

**Pull request merged**